packages/next/src/lib/verify-partytown-setup.ts

```diff
@@ -61,9 +62,45 @@ async function copyPartytownStaticFiles(
   await copyLibFiles(partytownLibDir)
 }
 
+async function verifyRootLayoutImport(appDir: string) {
+  const possibleLayoutFiles = [
+    'layout.ts',
+    'layout.tsx',
+    'layout.js',
+    'layout.jsx',
+  ]
+  let rootLayoutPath: string | undefined
+  let rootLayoutContent: string | undefined
+
+  for (const file of possibleLayoutFiles) {
+    const filePath = path.join(appDir, file)
+    if (await fileExists(filePath, FileType.File)) {
+      rootLayoutPath = filePath
+      rootLayoutContent = await readFile(filePath, 'utf8')
+      break
+    }
+  }
+
+  if (!rootLayoutPath || !rootLayoutContent) return
+
+  const hasPartytownComponent = /<Partytown/i.test(rootLayoutContent)
+
+  if (!hasPartytownComponent) {
+    Log.warn(
+      `Partytown is installed, but the Partytown component is not properly set up in your root layout file ${bold(rootLayoutPath)}. Make sure to import and use the Partytown component for proper setup with the App Router.`
+    )
+  }
+}
+
 export async function verifyPartytownSetup(
   dir: string,
-  targetDir: string
+  {
+    targetDir,
+    appDir,
+  }: {
+    targetDir: string
+    appDir?: string
+  }
```

Comment:
Not sure why changing the signature here. I'd go with either `verifyPartytownSetup(dir, targetDir, appDir)` which would be non-breaking for this fn (not sure if that matters) or `verifyPartytownSetup({dir, targetDir, appDir})` if this fn is not in some hot path, rather than mixing both

Reply:
You're absolutely right, keeping `dir` as a separate argument was unnecessary, not sure what I had in mind there. I’ve changed it to an object to maintain consistency with the other `verifyX` functions, which use objects when there is more than one argument